assh: fails to build on aarch64-darwin

### Describe the bug

The `assh` derivation fails to build on `aarch64-darwin`, macOS 12.0.1, due to an upstream package error.

### Steps To Reproduce

1. Be on `aarch64-darwin`, running Monterey
2. `nix-env -f '<nixpkgs>' -i assh`
3. Derivation fails to build.

### Expected behavior

The package should build and install successfully.


### Additional context

It seems like there's an upstream problem in one of the go package dependencies:

```console
vendor/github.com/shirou/gopsutil/cpu/cpu_darwin_cgo.go:13:5: error: 'TARGET_OS_MAC' is not defined, evaluates to 0 [-Werror,-Wundef-prefix=TARGET_OS_]
#if TARGET_OS_MAC
    ^
1 error generated.
```

It seems like the `CGO_CFLAGS` should be amended to have `-Wno-undef-prefix` to avoid the default LLVM/Clang 11 default compiler arguments of `-Werror,-Wundef-prefix=TARGET_OS_`.

Relevant issue: https://github.com/shirou/gopsutil/issues/976

#### Log output

```console
❯ nix-env -f '<nixpkgs>' -i assh
installing 'assh-2.12.0'
this derivation will be built:
  /nix/store/5dkm1yffnqgwh0zqfdww0my108s121fw-assh-2.12.0.drv
building '/nix/store/5dkm1yffnqgwh0zqfdww0my108s121fw-assh-2.12.0.drv'...
unpacking sources
unpacking source archive /nix/store/jq57bbxs3igpz9mb5jy70bdac3m7lzlh-source
source root is source
patching sources
updateAutotoolsGnuConfigScriptsPhase
configuring
building
Building subPackage .
encoding
internal/race
internal/unsafeheader
unicode/utf8
math/bits
unicode
runtime/internal/sys
unicode/utf16
sync/atomic
internal/cpu
runtime/internal/atomic
runtime/internal/math
container/list
crypto/internal/subtle
vendor/golang.org/x/crypto/cryptobyte/asn1
crypto/subtle
math
internal/nettrace
vendor/golang.org/x/crypto/internal/subtle
golang.org/x/sys/internal/unsafeheader
internal/bytealg
runtime/cgo
runtime
internal/reflectlite
sync
internal/singleflight
internal/testlog
math/rand
errors
sort
github.com/hashicorp/hcl/hcl/strconv
internal/oserror
path
vendor/golang.org/x/net/dns/dnsmessage
io
strconv
syscall
crypto/internal/randutil
hash
text/tabwriter
bytes
strings
crypto/hmac
hash/adler32
hash/crc32
vendor/golang.org/x/crypto/hkdf
golang.org/x/crypto/pbkdf2
crypto/rc4
crypto
encoding/base32
golang.org/x/text/transform
reflect
vendor/golang.org/x/text/transform
github.com/huandu/xstrings
bufio
regexp/syntax
html
internal/syscall/execenv
internal/syscall/unix
time
regexp
go.uber.org/zap/buffer
io/fs
internal/poll
context
golang.org/x/sys/unix
go.uber.org/zap/internal/bufferpool
golang.org/x/net/context
internal/fmtsort
encoding/binary
crypto/x509/internal/macos
os
encoding/base64
crypto/sha512
crypto/md5
crypto/sha1
crypto/sha256
vendor/golang.org/x/crypto/poly1305
crypto/cipher
crypto/ed25519/internal/edwards25519
encoding/pem
golang.org/x/crypto/scrypt
go.uber.org/zap/internal/exit
path/filepath
os/signal
vendor/golang.org/x/net/route
fmt
crypto/des
vendor/golang.org/x/crypto/chacha20
crypto/aes
github.com/mattn/go-isatty
golang.org/x/crypto/ssh/terminal
github.com/mattn/go-colorable
github.com/bugsnag/osext
github.com/spf13/afero/mem
io/ioutil
os/exec
vendor/golang.org/x/crypto/chacha20poly1305
net
go.uber.org/zap/internal/color
log
encoding/hex
flag
vendor/golang.org/x/crypto/curve25519
net/url
compress/flate
encoding/json
math/big
vendor/golang.org/x/text/unicode/norm
vendor/golang.org/x/net/http2/hpack
mime
vendor/golang.org/x/text/unicode/bidi
mime/quotedprintable
net/http/internal
github.com/docker/go-units
github.com/flynn/go-shlex
compress/gzip
github.com/mgutz/ansi
github.com/pkg/errors
encoding/csv
vendor/golang.org/x/text/secure/bidirule
text/template/parse
github.com/fsnotify/fsnotify
github.com/hashicorp/hcl/hcl/token
golang.org/x/text/unicode/norm
github.com/spf13/jwalterweatherman
github.com/subosito/gotenv
github.com/hashicorp/hcl/hcl/ast
github.com/hashicorp/hcl/hcl/scanner
github.com/hashicorp/hcl/json/token
gopkg.in/ini.v1
gopkg.in/yaml.v2
github.com/hashicorp/hcl/json/scanner
github.com/hashicorp/hcl/hcl/parser
vendor/golang.org/x/net/idna
go.uber.org/atomic
github.com/hashicorp/hcl/json/parser
github.com/hashicorp/hcl/hcl/printer
go.uber.org/multierr
github.com/hashicorp/hcl
go.uber.org/zap/zapcore
golang.org/x/time/rate
github.com/docker/docker/api/types/strslice
text/template
github.com/imdario/mergo
github.com/docker/libcompose/yaml
archive/zip
github.com/Masterminds/semver
database/sql/driver
github.com/mitchellh/reflectwalk
crypto/rand
crypto/elliptic
encoding/asn1
crypto/dsa
github.com/dustin/go-humanize
crypto/ed25519
crypto/rsa
github.com/pelletier/go-toml
github.com/deckarep/gosx-notifier
github.com/Masterminds/goutils
html/template
github.com/mitchellh/copystructure
vendor/golang.org/x/crypto/cryptobyte
crypto/x509/pkix
os/user
github.com/awalterschulze/gographviz/internal/token
encoding/gob
github.com/mattn/go-zglob/fastwalk
github.com/awalterschulze/gographviz/ast
github.com/awalterschulze/gographviz/internal/errors
github.com/awalterschulze/gographviz/internal/lexer
crypto/ecdsa
github.com/mattn/go-zglob
github.com/shirou/gopsutil/internal/common
github.com/spf13/cast
github.com/moul/flexyaml
github.com/shirou/gopsutil/cpu
github.com/shirou/gopsutil/mem
github.com/awalterschulze/gographviz/internal/parser
# github.com/shirou/gopsutil/cpu
vendor/github.com/shirou/gopsutil/cpu/cpu_darwin_cgo.go:13:5: error: 'TARGET_OS_MAC' is not defined, evaluates to 0 [-Werror,-Wundef-prefix=TARGET_OS_]
#if TARGET_OS_MAC
    ^
1 error generated.
github.com/awalterschulze/gographviz
github.com/haklop/gnotifier
github.com/shirou/gopsutil/net
github.com/mitchellh/mapstructure
vendor/golang.org/x/net/http/httpproxy
github.com/google/uuid
net/textproto
github.com/spf13/pflag
crypto/x509
vendor/golang.org/x/net/http/httpguts
mime/multipart
github.com/Masterminds/sprig
crypto/tls
github.com/spf13/cobra
net/http/httptrace
net/http
github.com/magiconair/properties
github.com/spf13/afero
go.uber.org/zap
moul.io/assh/v2/pkg/version
moul.io/assh/v2/pkg/templates
moul.io/assh/v2/pkg/utils
moul.io/assh/v2/pkg/ratelimit
github.com/spf13/viper
moul.io/assh/v2/pkg/hooks
moul.io/assh/v2/pkg/controlsockets
moul.io/assh/v2/pkg/config
moul.io/assh/v2/pkg/config/graphviz
error: builder for '/nix/store/5dkm1yffnqgwh0zqfdww0my108s121fw-assh-2.12.0.drv' failed with exit code 1;
       last 10 log lines:
       > go.uber.org/zap
       > moul.io/assh/v2/pkg/version
       > moul.io/assh/v2/pkg/templates
       > moul.io/assh/v2/pkg/utils
       > moul.io/assh/v2/pkg/ratelimit
       > github.com/spf13/viper
       > moul.io/assh/v2/pkg/hooks
       > moul.io/assh/v2/pkg/controlsockets
       > moul.io/assh/v2/pkg/config
       > moul.io/assh/v2/pkg/config/graphviz
       For full logs, run 'nix log /nix/store/5dkm1yffnqgwh0zqfdww0my108s121fw-assh-2.12.0.drv'.
```

### Notify maintainers

@zzamboni

### Metadata

```console
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 21.1.0, macOS 12.0.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.4`
 - channels(root): `"nixpkgs-21.11pre333180.98747f27ecf"`
 - channels(jperras): `"darwin, home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

```

--- a/pkgs/tools/networking/assh/default.nix
+++ b/pkgs/tools/networking/assh/default.nix
@@ -7,16 +7,16 @@
 
 buildGoModule rec {
   pname = "assh";
-  version = "2.12.0";
+  version = "2.12.1";
 
   src = fetchFromGitHub {
     repo = "advanced-ssh-config";
     owner = "moul";
     rev = "v${version}";
-    sha256 = "sha256-FqxxNTsZVmCsIGNHRWusFP2gba2+geqBubw+6PeR75c=";
+    sha256 = "1r3fny4k1crpjasgsp09qf0p3l9vg8c0ddbb8jd6qnqnwwprqfxd";
   };
 
-  vendorSha256 = "sha256-AYBwuRSeam5i2gex9PSG9Qk+FHdEhIpY250CJo01cFE=";
+  vendorSha256 = "14x7m900mxiwgbbxs56pdqsmx56c4qir5j4dz57bwr10rmh25fy4";
 
   doCheck = false;
 

